### PR TITLE
Added github action which fails if updates to our compiled json files are needed to be committed

### PR DESCRIPTION
**Pull request by ramesht007**
# Overview

When updates to our crawlers, aggregation script, or new manual translations are added, those updates aren't reflected in `threepio.py` until we re-run the crawlers & script and then commit the output.

# Solution

This PR has a Github Action which will run the crawlers and script and exit with an error if new changes are detected, reminding the user to commit the new compiled output.

### CODE REVIEW

**Comment by Nolski** [comment]
@ramesht007 I suspect this is because the location in which the action is being run is different. It might be worth trying out some debugging tools or looking into the `working-directory` property that you can use in github actions https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun

**Comment by ramesht007** [comment]
ok I will look into it

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Test
```
Comment (Nolski) [inline .github/workflows/test.yml]: Can you update this to something like `Run Crawlers`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Run all crawlers
+      run: chmod +x docs-crawler/docs/run_crawlers.sh
```
Comment (Nolski) [inline .github/workflows/test.yml]: It doesn't look like you're running the crawlers here but just changing the permissions on the file.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (Nolski) [inline .github/workflows/test.yml]: There's no `requirements.txt` as we're using [Poetry](https://python-poetry.org/). So we will either need to install the dependencies using poetry here, or switch to using pip (which I've been meaning to do).

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Run all crawlers
+      run: chmod +x docs-crawler/docs/run_crawlers.sh
+    - name: run aggregate_crawler
+      run: python docs-crawler/docs/aggregate_crawler_output.py
```
Comment (Nolski) [inline .github/workflows/test.yml]: Lastly, we want to do something here once the output has been generated. Either commit it to this branch, or fail if there are new changes. I'm okay with either but I think automatically committing the new changing might be more useful. What do you think?

**Review by Nolski — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (ramesht007) [reply .github/workflows/test.yml]: ok sure

File: .github/workflows/test.yml (hunk 2 above)
Comment (ramesht007) [reply .github/workflows/test.yml]: without it is giving permission denied, let me run it locally

File: .github/workflows/test.yml (hunk 3 above)
Comment (ramesht007) [reply .github/workflows/test.yml]: ok, but i saw requirement.txt in root

File: .github/workflows/test.yml (hunk 4 above)
Comment (ramesht007) [reply .github/workflows/test.yml]: ok, that's great idea

File: .github/workflows/test.yml (hunk 3 above)
Comment (ramesht007) [reply .github/workflows/test.yml]: so i should write
`pip install requirements.txt`

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    - name: Commit to repository
+      run: |
+        git config user.name "GitHub Action"
+        git config user.email "actions@users.noreply.github.com"
+        cd pythreepio/static
+        git add  mapped_commands_full.json
+        git commit -m "added final ouput"
+        git push origin master
+
```
Comment (Nolski) [inline .github/workflows/test.yml]: This section should be replaced with the `[ $(git status --short | wc -l) = 0 ]` we talked about in slack :)

Comment (ramesht007) [reply .github/workflows/test.yml]: ok

Comment (ramesht007) [reply .github/workflows/test.yml]: I was trying something new

Comment (ramesht007) [reply .github/workflows/test.yml]: I will make the changes

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        rm -r docs-crawler/docs/output/tfjs/1.5.1.json
+        rm -r docs-crawler/docs/output/torch/1.4.0.json
+    - name: Run all crawlers
+      run: |
+        cd docs-crawler/docs/
+        bash run_crawlers.sh
+    - name: run aggregate_crawler
+      run: python docs-crawler/docs/aggregate_crawler_output.py
+    - name: file cross check
+      run: (git status --short | wc -l) = 0
```
Comment (Nolski) [inline .github/workflows/test.yml]:
```suggestion
      run: [(git status --short | wc -l) = 0 ]
```

Comment (ramesht007) [reply .github/workflows/test.yml]: this dosen't run

Comment (ramesht007) [reply .github/workflows/test.yml]: take a look at # revert above

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        rm -r docs-crawler/docs/output/tfjs/1.5.1.json
+        rm -r docs-crawler/docs/output/torch/1.4.0.json
+    - name: Run all crawlers
+      run: |
+        cd docs-crawler/docs/
+        bash run_crawlers.sh
+    - name: run aggregate_crawler
+      run: python docs-crawler/docs/aggregate_crawler_output.py
+    - name: file cross check
+      run: [(git status --short | wc -l) = 0 ]
```
Comment (Nolski) [inline .github/workflows/test.yml]:
```suggestion
      run: "[(git status --short | wc -l) = 0 ]"
```

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        rm -r docs-crawler/docs/output/tfjs/1.5.1.json
+        rm -r docs-crawler/docs/output/torch/1.4.0.json
+    - name: Run all crawlers
+      run: |
+        cd docs-crawler/docs/
+        bash run_crawlers.sh
+    - name: run aggregate_crawler
+      run: python docs-crawler/docs/aggregate_crawler_output.py
+    - name: file cross check
+      run: "[(git status --short | wc -l) = 0 ]"
```
Comment (Nolski) [inline .github/workflows/test.yml]:
```suggestion
      run: "[ $(git status --short | wc -l) = 0 ]"
```

File: .github/workflows/test.yml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        python-version: [3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run:
```
Comment (Nolski) [inline .github/workflows/test.yml]:
```suggestion
      run: |
```

**Comment by ramesht007** [comment]
tried experimenting with something but messed in my local repo and now I will have to refork the repo and then give a new PR. Sorry for any inconvenience.

**Pull request closed without merging**